ccm_web/server/src/app.module.ts

```diff
@@ -24,6 +27,19 @@ const logger = baseLogger.child({ filePath: __filename })
       ignoreEnvFile: true,
       isGlobal: true
     }),
+    ServeStaticModule.forRootAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: (configService: ConfigService<Config, true>) => ([{
+        rootPath: (
+          path.join(
+            path.join(__dirname, '..', '..'),
+            configService.get('server.isDev', { infer: true }) ? path.join('dist', 'client') : 'client'
+          )
+        ),
+        exclude: ['/api*', '/canvas*', '/lti*', '/auth*']
```

Comment:
does that mean that if we have a static react route with any of the names above might be conflicting? Like if we have react page as `https:/dff.ngrok.com/canvas-section` will be blocked. 

What you have done is cool and seems to working, just trying to wrap my head around with this PR

Reply:
This is a good point, hm...

Yes, I think `canvas-section` would conflict. I think for `api`, `canvas`, and `auth`, it's probably safe to add `/` at the end of each of them. I'll have to check whether `/lti/*` introduces any weird behavior.

Reply:
It also looks like if I remove this entirely, the existing API routes still work, they just all fall through to the Not Found page. Maybe this is fine behavior. Currently anything not defined after these route beginnings result in a standard JSON `express` error.

Reply:
OK

Reply:
I decided to add the `/` at the end of each of them so something like `canvas-sections` or `lti-error` would still fallthrough. I think excluding the routes is still a good practice, since anything after these prefixes **should** be an API route and show the appropriate type error instead of loading the static.

Interestingly, with the latest commit (), a route like `/canvas` would still be excluded. It looks like this is because there is an appended slash when the `ServeStaticModule` check whether a route is excluded: https://github.com/nestjs/serve-static/blob/f9c7be3365359d86395f33b6c8c7d8aa1269de9d/lib/utils/is-route-excluded.util.ts#L12-L16